### PR TITLE
Support yarn.lock

### PR DESCRIPTION
**Pull request by methyl**
This PR adds support for parsing yarn.lock instead of npm-shrinwrap.json.

It was pretty trivial as yarn has module to parse the file to json format, which seems to be pretty much compatible with shrinkwrap.

### CODE REVIEW

**Comment by methyl** [comment]
It turns out yarn is not babelifying their modules. I will need to figure out another way to pull its parser in, do you have any ideas how to do that properly? None of my seems ideal.

**Comment by methyl** [comment]
Added PR to yarn to support es5, let's keep fingers crossed they can accept it: https://github.com/yarnpkg/yarn/pull/2275

**Comment by methyl** [comment]
So it turns out they won't be able to accept that PR adding ES5 support. Do you think we can use my yarn fork for now?

**Comment by mike-engel** [comment]
I wonder if it's worth supporting node < v4, as v4 is the lowest supported version of node now? I'm sure the node security team has some thoughts on that and maybe they're not ready to drop support for node versions less than 4. Maybe a new, major version of nsp?

**Comment by omeid** [comment]
Anyone who cares about security wouldn't use an unsupported platform, or so one would hope.


PS. I know that "in theory, theory and practice are the same. In practice, they are not.", but still, most people who would use nsp would care at least enough to stay on a support version.

**Comment by mike-engel** [comment]
@methyl what would this PR look like with node >= v4 support? Looks like the PR is still using your fork

**Comment by methyl** [comment]
I will submit new one tomorrow. I like the idea of not supporting old node
versions.

śr., 11.01.2017, 23:08 użytkownik Mike <notifications@github.com> napisał:

**Comment by methyl** [comment]
This is how it would look like. @nlf any comments on that?

**Review by mike-engel — APPROVED** [review]

**Comment by frenchi** [comment]
We've just migrated over to yarn, and this has caused our nodesecurity scans to fail. Following with keen interest!

Has there been a decision on node <= v4 support?

(P.S. @evilpacket & @jlamendo good to meet you guys at DEFCON a couple of years back - didn't see you last year)

**Comment by evilpacket** [comment]
@frenchi 👋 - @nlf will officially comment but I know that yarn support is on our list however we do have to make some API tweaks to make it officially go and have a few things in the queue in front of it.

**Comment by nlf** [comment]
this is a great start for offline mode :+1:

we do need to decide if we're officially dropping support for node < 4 before i merge it, however. for right now, i'm going to keep this open. i do want to say *THANK YOU* for doing this and including tests and everything. it's super appreciated!

**Comment by devinus** [comment]
Any updates on this elsewhere?

**Comment by nlf** [comment]
ok, short term plan - we'll be dropping support for node versions less than 4 in the near term. once that happens this support will be added. sorry for the delay folks, but sit tight, this support will be added soon!

**Comment by zcei** [comment]
@nlf something we can support you with? Only reference to node version I found was in `drone` config. But maybe there's something within the advisories that needs to be filtered out, as it only applies to Node < 4?

**Comment by methyl** [comment]
Any updates on this one?

**Comment by timja** [comment]
@nlf sorry to poke again, but any updates?

**Comment by martywins** [comment]
@methyl Thanks for taking the initiative to add support for yarn.

I would LOVE this to be merged asap.  I'd like to use NSP but my project team use yarn, so this is effectively stopping us using NSP.

@nlf @mike-engel Any updates?

**Comment by methyl** [comment]
https://nodejs.org/en/blog/vulnerability/july-2017-security-releases/ This means everyone under Node 4 is vulnerable, I think it's good time to drop the support finally.

**Comment by nlf** [comment]
I agree! Work is in progress :)

**Comment by jborrey** [comment]
@methyl how is this going? Still hoping to merge this?

**Comment by hermanbanken** [comment]
What is the status of this? The merge conflicts can be resolved (see https://github.com/hermanbanken/nsp/tree/pr/142)

**Comment by hermanbanken** [comment]
Ah, we should use `@yarnpkg/lockfile`. I think I'm gonna submit a new PR...

**Comment by nlf** [comment]
see https://github.com/nodesecurity/nsp/pull/187#issuecomment-334624079

**Pull request closed without merging**
